src/js/testem-component.js

```diff
@@ -0,0 +1,531 @@
+            args:     ["http://localhost:%port", { port: "{that}.options.coveragePort" }]
+        }
+    },
+    coverageDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "coverage", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    reportsDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "reports", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    testemDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "user_data_dir", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    instrumentSource: true,
+    // Code to be served must live under the cwd in order to work with Testem.
+    instrumentedSourceDir: {
+        expander: {
+            funcName: "gpii.testem.resolveSafely",
+            args:     ["{that}.options.testemOptions.cwd", "instrumented"]
+        }
+    },
+    generateCoverageReport: true,
+    sourceDirs: [],
+    testPages:   [],
+    serveDirs:  [],
+    members: {
+        generatedOptions: {
+            routes: {}
+        }
+    },
+    events: {
+        initialCleanupComplete: null,
+        constructFixtures: null,
+        sourceInstrumented: null,
+        safeToConstructFixtures: {
+            events: {
+                initialCleanupComplete: "initialCleanupComplete",
+                constructFixtures:      "constructFixtures",
+                sourceInstrumented:     "sourceInstrumented"
+            }
+        },
+        onExpressStarted: null,
+        onFixturesConstructed: {
+            events: {
+                onExpressStarted: "onExpressStarted"
+            }
+        },
+        stopFixtures: null,
+        onExpressStopped: null,
+        onFixturesStopped: {
+            events: {
+                onExpressStopped: "onExpressStopped"
+            }
+        },
+        createReports: null,
+        onCoverageReportComplete: null,
+        onReportsComplete: {
+            events: {
+                onCoverageReportComplete: "onCoverageReportComplete"
+            }
+        },
+        finalCleanup: null,
+        finalCleanupComplete: null
+    },
+    testemOptions: {
+        framework:   "qunit",
+        report_file: {
+            expander: {
+                funcName: "gpii.testem.resolveSafely",
+                args:     ["{that}.options.reportsDir", "report.tap"]
+            }
+        },
+        cwd: process.cwd(),
+        user_data_dir: "{that}.options.testemDir",
+        on_start: "{that}.handleTestemStart",
+        on_exit:  "{that}.handleTestemExit",
+        test_page: "{that}.options.testPages",
+        proxies: {
+            "/coverage": {
+                "target": "{that}.options.coverageUrl"
+            }
+        }
+    },
+    invokers: {
+        "handleTestemStart": {
```

Comment:
I'd really like to see these embodied as namespaced listeners to an event rather than invokers. It seems quite likely that the "responsible" user might want to insert themselves into this workflow.
We can always embody the "failsafe" part of the workflow into a listener with priority "last" - this doesn't lead to any increased risk than the one you already have within gpii.testem.init - since if an exception is thrown during the processing of listeners to "onFixturesConstructed", the callback will be dangling in any case. And I think users of the design will appreciate there being Infusion events named by direct correspondence to the underlying testem ones.